### ISSUE TITLE
Multiple issues related to privacy settings

### ISSUE THREAD

**Issue by profjb58** [SWEngGroup4/scranplan#112]
In descending order...
_Apologies for not outlining these initially before merge, they where quite hard to find and involved making 2 accounts._

There are other issues as well currently in master but i've resolved them with the 'bug+design fixes' branch. Might be worth doing some pair programming to fix this. Certainly could speed things up a bit or if yer short for time I could have a stab.

- Privacy options no longer sync as this had been commented out in the edit profile page and so now displays incorrectly when switching between private/public profiles
- A user who dosen't have a private account (IE everything is public) cannot hide there Feed or Recipes.
- When private profile is disabled you can't view the image for your own profile.
- With a private profile disabled when changing any privacy setting it updates the 'privatePrivacy' collection. It should update 'publicPrivacy'.
- When switching to the 'photos' tab after loading another users profile the app crashes. Seems to be an issue with a null object reference. Code for the crash for this one is below.

Line it crashes at:
![image](https://user-images.githubusercontent.com/46009144/83209100-56dc7c00-a14f-11ea-8fdb-80fdffe0d8ec.png)

```E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.group4sweng.scranplan, PID: 22436
    java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
        at com.group4sweng.scranplan.PublicProfile$4.onTabSelected(PublicProfile.java:302)
        at com.google.android.material.tabs.TabLayout.dispatchTabSelected(TabLayout.java:1734)
        at com.google.android.material.tabs.TabLayout.selectTab(TabLayout.java:1727)
        at com.google.android.material.tabs.TabLayout.selectTab(TabLayout.java:1687)
        at com.google.android.material.tabs.TabLayout$Tab.select(TabLayout.java:2080)
        at com.google.android.material.tabs.TabLayout$TabView.performClick(TabLayout.java:2269)
        at android.view.View.performClickInternal(View.java:7102)
        at android.view.View.access$3500(View.java:801)
        at android.view.View$PerformClick.run(View.java:27336)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)```

**Comment by nathanbillis** [comment]
How can this be both a major and minor bug?

**Comment by louisifn** [comment]
Privacy options no longer sync as this had been commented out in the edit profile page and so now displays incorrectly when switching between private/public profiles -> what specifically doesn't sync? If a user has a private account then different items load. I'm struggling to recreate this? All items commented out were commented out on purpose, the class was too maze-like to rewrite sections of code and we didn't have time.

**Comment by louisifn** [comment]
With a private profile disabled when changing any privacy setting it updates the 'privatePrivacy' collection. It should update 'publicPrivacy'. -> no it should not. The two items load differently, private has more options, if a user has a public account it means they do not need public options as their account is not private. Thus they retain all privacy options but using a full profile load.

**Comment by louisifn** [comment]
When switching to the 'photos' tab after loading another users profile the app crashes. Seems to be an issue with a null object reference. Code for the crash for this one is below. -> good find I shall fix this one asap

**Comment by louisifn** [comment]
A user who dosen't have a private account (IE everything is public) cannot hide there Feed or Recipes. -> this is an interesting one. I disagree with hiding any recipes in the home page so im going to avoid making any changes here. As with feed, I feel this is different to allowing using to see all posts, we could either disable the option for users to hide their posts to followers, or we can leave it so followers can see three most recent posts. Disabling this would disable the ability for users to make posts on feed? Or even see the feed. I think this issue is a dead end and we would be best off just disabling the ability to hide posts from followers, far less admin.

**Comment by louisifn** [comment]
When private profile is disabled you can't view the image for your own profile. -> will look into this

**Comment by profjb58** [comment]
I've just lumped the bugs into one since there under the same kind of category and labelled major and minor since some are major and some are minor. I can split them up if needed I just didn't want to make 5 new github issues.

- The sync visibility does seem to be used. Not sure why I brang that up as an issue. So a non-issue.

- Not sure why the private profile collection is being updated though for when a users selects to have a public profile. (Ie turns private profile off).  From what you said it seems intentional which is fine but because the local public profile (IE when a user views there own profile) only ever loads the users 'public' profile collection to see what it looks like this needs to be adjusted.
It's a very quick fix though and I can work on that.

- I added the ability to hide recipes since I think originally we had the idea of 'private' recipes or recipes that only the user themselves could see without sharing and this was the option that enabled or disabled it, since this feature hasn't been added it's probably worth getting rid of the 'recipes' filter altogether. At this point it's probably worth updating the privacy collection with 2 unused variables so i'll get to that.
The feed however is still an issue. The option to disable the feed is still present in the edit profile settings when switching to a public account but does nothing.

I also need to figure out what the public profile is for (local public profile, the one navigated to through the sidebar).
Originally it was meant to just display what all public users could see so that the user was aware of whats hidden or not to the public. Since the addition of the feed this gets a bit more complicated.
An approach to fix this would to be have a switch which switches between what the public can see and what can be seen privately if a private profile is enabled.
Displaying the private profile only has it's own issues with not making it clear whats hidden to certain users so this approach wouldn't really work either.

**Comment by nathanbillis** [comment]
Yeah that’s fine I understand, the problem arises when we fix them no in
order and not at the same time meaning we can’t close this ticket and work
out what’s been done.

**Comment by profjb58** [comment]
Yeah thats true. I think since there all related to the same issue of privacy options and given the short amount of time we have left they will all have to get fixed as one anyway though.

**Comment by profjb58** [comment]
All the above except the image tab switch issue are fixed in #125 now.

**Comment by profjb58** [comment]
Image tab switch issue is now fixed in #125
